source/slang/slang.cpp

```diff
@@ -2373,9 +2410,28 @@ DeclRef<Decl> ComponentType::findDeclFromStringInType(
         result = declRefExpr->declRef;
     }
 
+    if (auto genericDeclRef = result.as<GenericDecl>())
+    {   
+        result = createDefaultSubstitutionsIfNeeded(
+            astBuilder, &visitor, DeclRef(genericDeclRef.getDecl()->inner));
+        result = substituteDeclRef(SubstitutionSet(genericDeclRef), astBuilder, result);
+    }
+
     return result;
 }
 
+bool ComponentType::isSubType(Type* subType, Type* superType)
+{
+    SharedSemanticsContext sharedSemanticsContext(
```

Comment:
At somepoint it may make sense to hold a semantic visitor on the linkage, because it will become extensive to rebuild the type inheritance list for all referenced types everytime `isSubType` is called. These are cached on the semantic visitor and we should probably reuse it to speed up these reflection queries.